### PR TITLE
Feat/editable participant

### PR DESCRIPTION
**Pull request by dontry**
This PR enables users to edit participant name:

1. Add a new member `participantPositions` to `Pariticipants` class. `participantPositions` is a map where to keep track of participant names and their positions.
2. Record participant's position when traversing the parseTree. 
3. Create a new component `PariticipantLabel` for editing label names and updating the participant instances in code

4. Disable edit in static render mode

https://github.com/mermaid-js/zenuml-core/assets/3481791/7e94e357-10fb-4f17-8915-ebc783fb9225

### CODE REVIEW

**Comment by MrCoder** [comment]
In the following case, the expected result should be
```
A as "A is a service 1"
A.method
```

https://github.com/mermaid-js/zenuml-core/assets/169417/a9461ce6-0c38-425f-bed2-6587c43d11b5

**Review by MrCoder — CHANGES_REQUESTED** [review]
It seems that it does not handle participant alias correctly.
The syntax is `ParticipantA as ParticipantA_Alias`.

**Comment by dontry** [comment]
> It seems that it does not handle participant alias correctly. The syntax is `ParticipantA as ParticipantA_Alias`.

Fixed

**Comment by MrCoder** [comment]
First part is on the main branch where `new E` works as expected. The second part is the PR branch participant is not displayed with `new E`.
https://github.com/mermaid-js/zenuml-core/assets/169417/969b8bda-7153-4591-9b90-783bb2f012dd

File: src/parser/ToCollector.js [hunk 1]
```diff
@@ -46,21 +64,29 @@ ToCollector.enterParticipant = onParticipant;
 let onTo = function (ctx) {
   if (isBlind) return;
   let participant = ctx.getFormattedText();
-  participants.Add(participant);
+  const participantInstance = participants.Get(participant);
```
Comment (MrCoder) [inline src/parser/ToCollector.js:67]: Same logic needs to be implemented in `enterCreation`?

File: src/parser/ToCollector.js [hunk 2]
```diff
@@ -46,21 +64,29 @@ ToCollector.enterParticipant = onParticipant;
...
   let participant = ctx.getFormattedText();
-  participants.Add(participant, true);
+  participants.Add(participant, true, ctx.start.start, ctx.stop.stop + 1);
 };
 
 ToCollector.enterCreation = function (ctx) {
   if (isBlind) return;
   const participant = ctx.Owner();
-  participants.Add(participant);
+  const ctor = ctx.creationBody().construct();
```
Comment (MrCoder) [inline src/parser/ToCollector.js:88]: `ctor` could be null when the user is typing.
For example, the user pauses at
```
A.method {
  new 
}
```
In such case, the diagram is rendered as
image

File: test/unit/parser/to-collector.spec.js [hunk 3]
```diff
@@ -127,76 +132,106 @@ function getParticipants(code, withStarter) {
...
+      let participants = getParticipants("@Starter(X) IA a = new A()", true);
+      expect(participants.Size()).toBe(2);
+      expect(participants.Get("X").width).toBeUndefined();
+      expect(participants.Get("a:A").width).toBeUndefined();
+      expect(participants.GetPositions("X").size).toBe(1);
+      expect(participants.GetPositions("X")).toEqual(new Set(["[9,10]"]));
+      expect(participants.GetPositions("a:A")).toEqual(new Set(["[23,24]"]));
+    });
+    test("seqDsl should treat method call as a participant - creation & assignment", () => {
+      let participants = getParticipants("a = new A()", true);
```
Comment (MrCoder) [inline test/unit/parser/to-collector.spec.js:189]: This test is a dup of "seqDsl should treat creation as a participant - assignment".

Comment (dontry) [reply test/unit/parser/to-collector.spec.js]: Removed

File: src/parser/ToCollector.js (hunk 2 above)
Comment (dontry) [reply src/parser/ToCollector.js]: Fixed by adding optional chaining.

**Comment by dontry** [comment]
Address the issue of calling a method after creating a new instance and assigning to a new variable
https://github.com/mermaid-js/zenuml-core/assets/3481791/f59ee626-9063-4aa9-8744-ca21a3fc75d0

**Review by MrCoder — APPROVED** [review]

**Pull request merged**